src/status_im2/contexts/wallet/events.cljs

```diff
@@ -0,0 +1,19 @@
+(ns status-im2.contexts.wallet.events
+  (:require [utils.re-frame :as rf]
+            [taoensso.timbre :as log]))
+
+(rf/defn get-wallet-tokens
+  {:events [:wallet-2/get-wallet-tokens]}
+  [_ accounts]
+  {:json-rpc/call [{:method     "wallet_getWalletToken"
+                    :params     [(map :address accounts)]
+                    :on-success #(rf/dispatch [:wallet-2/get-wallet-tokens-success %])
+                    :on-error   #(log/info "failed " %)}]})
+
+(rf/defn get-wallet-tokens-success
+  {:events [:wallet-2/get-wallet-tokens-success]}
+  [{:keys [db]} data]
+  {:db (assoc db
+              :wallet-2/tokens          data
+              :wallet-2/tokens-loading? false)})
+
```

Comment:
without putting too much of a blocker on this pr, we should keep a reasonable eye on @ilmotta 's pr and if there is some consensus in a timely manner than we should update the new wallet api's to use that approach instead

Reply:
Done